### PR TITLE
Link Preview Embed

### PR DESCRIPTION
**Pull request by vasfvitor**
Hello, this aims to solve #30

Right now it's called `astro-embed-og`, to be renamed to `astro-link-preview` as mentioned in the issue.

TL:DR - How to go about styling and matching urls to the integration, right now it has a conditional class and matches any url.

I'm not sure how go about the best way to expose styles to allow end user customize without having to add `!important`. ~~Could just remove all styles from the card and provide current styling as an example. Could make conditional classes~~.  Any ideas?

yadayada from me

> Initially It would retrieve all metatags, as OpenGraph, Twitter and a custom, if provided. Then it would utilize only those set by the user:
> `<OpenGraph source="og" url="https://astro.build/blog/welcome-world/" />`
> `<OpenGraph source="twitter" url="https://astro.build/blog/welcome-world/" />`
> 
> But then I removed to simplify the code and because usually they are all the same data, except for custom, maybe. I was trying to mimick those cards: https://www.opengraph.xyz/url/https%3A%2F%2Fastro.build%2Fblog%2Fwelcome-world%2F
> 

And now, how it would go to match urls in the integration? if needed

Still have to test tests.

I utilized the cache code from `safeGet()` but as with my current knowledge, I'm not sure it's needed or if it's being utilized properly.

Update (previous version images is commented out):
Preview:

Thank you.

### CODE REVIEW

**Review by delucis — APPROVED** [review]
Hi @vasfvitor! So sorry it took me long to get back to you about this PR.

I hope you don’t mind that I jumped in over this weekend to update the branch and polish up your work to get it ready for release. Really love this idea!

> TL:DR - How to go about styling and matching urls to the integration, right now it has a conditional class and matches any url.

Re: styling, I focused on only providing a good structure and layout styles in the card. Now that we have a proper docs website, I documented how to add custom styles manually instead of trying to offer an opinionated styled version.

Re: matching URLs, for now I skipped the idea of allowing users to specify a custom pattern for matching URLs for this component, although I think it could be a helpful future feature. However, I did a new `services` option to the integration, which allows users to disable the `LinkPreview` matching if they don’t want it but want to keep the other existing services.

> I utilized the cache code from `safeGet()` but as with my current knowledge, I'm not sure it's needed or if it's being utilized properly.

I refactored this a bit to avoid duplicating the `safeGet()` logic — but the basic idea seemed solid!

> Thank you.

Thank you! So cool to have this component added and your foundational work was really good 🙌
